org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/contentassist/CompletionProposalReplacementProvider.java

```diff
@@ -607,7 +608,27 @@ private final boolean canAutomaticallyAppendSemicolon(CompletionProposal proposa
 			onlyAppendArguments= false;
 		}
 		if (onlyAppendArguments || shouldAppendArguments(typeProposal, trigger)) {
-			String[] typeArguments = computeTypeArgumentProposals(typeProposal);
+			String[] typeArguments = null;
+			if (originalProposal instanceof InternalCompletionProposal) {
+				char[] declSig = ((InternalCompletionProposal) originalProposal).getDeclarationSignature();
```

Comment:
hold on, `InternalCompletionProposal.getDeclarationSignature()` doesn't always contains type arguments info, which can lead to missing `<>` for some cases.